scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,359 @@
+    MTsat = 100*(((Aapp*acq_parameters[0][1]*acq_parameters[0][0]/T1app)/cPD2)-(acq_parameters[0][0]/T1app)-(acq_parameters[0][1]**2)/2)
+
+    return MTR, MTsat
+
+
+def threshold_ihMT_maps(computed_map, contrast_maps, segment_tissue,
+                        lower_threshold, upper_threshold, idx_contrast_list):
+    """
+    Remove NaN and apply different threshold based on
+       - maximum and minimum threshold value
+       - combination of specific contrasts maps
+    Multiple data by sum of T1 probability maps (GM+WM+CSF)
+
+    Parameters
+    ----------
+    computed_map        3D-Array data.
+                        Myelin map (ihMT or non-ihMT maps)
+    contrast_maps       List of 3D-Array. File must containing the
+                        6 contrasts maps.
+    segment_tissue      List of path to tissue probability maps from
+                        T1 segmentation
+    lower_threshold     Value for low thresold <int>
+    upper_thresold      Value for up thresold <int>
+    idx_contrast_list   List of indexes of contrast maps corresponding to
+                        that of input contrast_maps ex.: [0, 2, 5]
+                        Altnp = 0; Atlpn = 1; Reference = 2; Negative = 3;
+                        Positive = 4; T1weighted = 5
+    Returns
+    ----------
+    Thresholded matrix in 3D-array.
+    """
+    # Remove NaN and apply thresold based on lower and upper value
+    computed_map[np.isnan(computed_map)] = 0
+    computed_map[np.isinf(computed_map)] = 0
+    computed_map[computed_map < lower_threshold] = 0
+    computed_map[computed_map > upper_threshold] = 0
+
+    # Load and apply sum of T1 probability maps on myelin maps
+    mask = []
+    for img in segment_tissue:
+        load_image = nib.load(img)
+        mask.append(load_image.get_fdata(dtype=np.float64))
+    mask_data = mask[0]+mask[1]+mask[2]
+    computed_map[np.where(mask_data == 0)] = 0
+
+    # Apply threshold based on combination of specific contrasts maps
+    for idx in idx_contrast_list:
+        computed_map[contrast_maps[idx] == 0] = 0
+
+    return computed_map
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_output_dirs_exist_and_empty(parser, args,
+                                       os.path.join(args.out_dir,
+                                                    'Contrats_MT_maps'),
+                                       os.path.join(args.out_dir,
+                                                    'ihMT_native_maps'),
+                                       create_dir=True)
+
+    # Merged all echoes path in a list
+    maps = [args.in_altnp, args.in_altpn, args.in_mtoff, args.in_negative,
+            args.in_positive, args.in_t1w]
+
+    # Check the number of echoes for each contrast and json file
+    for contrast in maps:
```

Comment:
Instead of this do something like this

```
length = len(list1)
if all(len(lst) != length for lst in maps):
parser.error('Not the same number of echoes per map')
 ```